griptape/loaders/base_text_loader.py

```diff
@@ -43,12 +45,14 @@ def _text_to_artifacts(self, text: str) -> list[TextArtifact]:
 
         chunks = self.chunker.chunk(text) if self.chunker else [TextArtifact(text)]
 
-        if self.embedding_driver:
-            for chunk in chunks:
+        for chunk in chunks:
+            if self.embedding_driver:
```

Comment:
Why the swap? Isn't it cheaper to check if the embedding driver is set once vs every iteration?

Reply:
It was hard for me to make sense of the code during the initial read and I realized it's because we had the same for loop twice. I just combined them into one. Conditional checks are very cheap performance-wise, so I think it's a reasonable tradeoff.